### PR TITLE
disco 0.5.4

### PR DESCRIPTION
**Pull request by ilovezfs**
Upstream PR discoproject/disco#637 gets it working with >= Erlang 18.1

### CODE REVIEW

File: Formula/disco.rb [hunk 1]
```diff
@@ -11,12 +11,26 @@ class Disco < Formula
...
+  end
+
+  # Upstream PR adding Erlang 18.1 (passed their Travis Erlang 18.1 CI)
+  patch do
+    url "https://github.com/discoproject/disco/pull/632.patch"
+    sha256 "17f52a7e942b3913b2be2b362d29fd308dc253af211fa82cae38cddb0678f525"
+  end
+
+  # Modifies config for single-node operation.
   patch :DATA
```
Comment (mikemcquaid) [inline Formula/disco.rb]: Would be good to submit this upstream or otherwise remove it.

Comment (mikemcquaid) [inline Formula/disco.rb]: Or just `inreplace` it instead if it's configuration.

Comment (ilovezfs) [inline Formula/disco.rb]: @mikemcquaid I'm not sure it's intent but it seems to only be about configuration. I believe @apjanke added the original patch, so maybe he can shed some light.

**Comment by ilovezfs** [comment]
This is the diff to the PR here if we want to go with the "competing" upstream patch that is newer:
```
diff --git a/Formula/disco.rb b/Formula/disco.rb
index 24055bf..30dfd84 100644
--- a/Formula/disco.rb
+++ b/Formula/disco.rb
@@ -18,16 +18,9 @@ class Disco < Formula
   depends_on "simplejson" => :python if MacOS.version <= :leopard
   depends_on "libcmph"
 
-  # .travis.yml changed after the tag but before the Erlang 18.1 patch
   patch do
-    url "https://github.com/discoproject/disco/commit/14e726f1627ad3e7f1767722f36fb966e244185b.patch"
-    sha256 "94b23cd201f10b141bb839e4169a4648d4d6a10d95dccee243aad15558e1ae41"
-  end
-
-  # Upstream PR adding Erlang 18.1 (passed their Travis Erlang 18.1 CI)
-  patch do
-    url "https://github.com/discoproject/disco/pull/632.patch"
-    sha256 "17f52a7e942b3913b2be2b362d29fd308dc253af211fa82cae38cddb0678f525"
+    url "https://github.com/discoproject/disco/pull/637.patch"
+    sha256 "97336a33ed115e476ec3aba0692236c4a9e22f17605fd9057ad415772a0f73ec"
   end
 
   # Modifies config for single-node operation.

```
Not sure if anyone has opinions about https://github.com/discoproject/disco/pull/632 vs https://github.com/discoproject/disco/pull/637 but they both seem to work fine.

Also, the formula does this right now:
```
    # Disco's "rebar" build tool refuses to build unless it's in a git repo, so
    # make a dummy one.
    system "git init && git add master/rebar && git commit -a -m 'dummy commit'"

```

Would it be better to just build directly from the tag via git (like `osquery` does) to get rid of this, or just leave it? (It was successful when I tried it that way.)

**Comment by mikemcquaid** [comment]
Either patch seems fine to me.

> Would it be better to just build directly from the tag via git (like osquery does) to get rid of this, or just leave it? (It was successful when I tried it that way.)

Yeh, building from the Git tag makes more sense to me :+1:

File: Formula/disco.rb (hunk 1 above)
Comment (apjanke) [inline Formula/disco.rb]: This formula was basically the first Homebrew thing I ever worked on, so my memory's a little dim, but: Yeah, it's just configuration: Homebrew mostly targets people with standalone PCs, not server farms, so this changes the defaults to work for a single machine. The `caveats` go with it.

`inreplace` would be better than a patch anyway. More readable and robust to version changes. I think I would've done it that way in the first place if I had been more experienced when I wrote this. You can probably make it a one-liner, too, replacing `DDFS_.*_REPLICAS *= * (\d)` with 1.

**Comment by ilovezfs** [comment]
@mikemcquaid @apjanke @tdsmith PR refreshed.

File: Formula/disco.rb [hunk 2]
```diff
@@ -11,72 +12,83 @@ class Disco < Formula
...
+    ENV.deparallelize { system "make", "install" }
+    prefix.install %w[contrib doc examples]
+
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  def post_install
+    # Fix the config file to point at the linked files, not into the Cellar.
+    if File.read(etc/"disco/settings.py").include?("Cellar/disco/")
+      inreplace etc/"disco/settings.py", "Cellar/disco/"+version+"/", ""
```
Comment (mikemcquaid) [inline Formula/disco.rb]: does using `prefix` and `HOMEBREW_PREFIX` not work here?

**Pull request closed without merging**